### ISSUE TITLE
goveralls crashes with panic: runtime error: index out of range

### ISSUE THREAD

**Issue by porridge** [mattn/goveralls#86]
It seems that gover sometimes produces a garbage profile file, and goveralls crashes in a less than friendly manner when this happens:

```
$ goveralls -coverprofile=gover.coverprofile -service travis-ci
[...]
14:50:43 gocover.go:104: converting profile k8s.io/ingress/controllers/nginx/pkg/template/configmap.go with 113 lines and 113 startline and 113 endline
14:50:43 gocover.go:104: converting profile k8s.io/ingress/controllers/nginx/pkg/template/configmap.go with 113 lines and 113 startline and 113 endline
14:50:43 gocover.go:104: converting profile k8s.io/ingress/controllers/nginx/pkg/template/configmap.go with 113 lines and 113 startline and 113 endline
14:50:43 gocover.go:104: converting profile k8s.io/ingress/controllers/nginx/pkg/template/configmap.go with 113 lines and 113 startline and 113 endline
14:50:43 gocover.go:104: converting profile k8s.io/ingress/controllers/nginx/pkg/template/configmap.go with 113 lines and 116 startline and 118 endline
panic: runtime error: index out of range

goroutine 1 [running]:
panic(0x6e3480, 0xc420012190)
	/home/porridge/.gvm/gos/go1.7.3/src/runtime/panic.go:500 +0x1a1
main.toSF(0xc420456000, 0x35, 0x35, 0x35, 0x35, 0x0, 0x0, 0x5340e9)
	/home/porridge/Desktop/coding/go/src/github.com/mattn/goveralls/gocover.go:106 +0x71d
main.parseCover(0x7ffcc1379d5b, 0x12, 0x697370, 0x43006c, 0xc4200517f0, 0xc4200517f0, 0x40000000)
	/home/porridge/Desktop/coding/go/src/github.com/mattn/goveralls/gocover.go:122 +0x119
main.getCoverage(0x738a3e, 0xf, 0x0, 0x0, 0x0)
	/home/porridge/Desktop/coding/go/src/github.com/mattn/goveralls/goveralls.go:114 +0xc3d
main.process(0x0, 0x0)
	/home/porridge/Desktop/coding/go/src/github.com/mattn/goveralls/goveralls.go:237 +0x2ea
main.main()
	/home/porridge/Desktop/coding/go/src/github.com/mattn/goveralls/goveralls.go:327 +0x26
```

The above is from goveralls with the following patch:

```
--- a/gocover.go
+++ b/gocover.go
@@ -100,6 +100,8 @@ func toSF(profs []*cover.Profile) ([]*SourceFile, error) {
                }
 
                for _, block := range prof.Blocks {
+                      log.Printf("converting profile %s with %d lines and %d startline and %d endline",
+                                  prof.FileName, len(sf.Coverage), block.StartLine, block.EndLine)
                        for i := block.StartLine; i <= block.EndLine; i++ {
                                count, _ := sf.Coverage[i-1].(int)
                                sf.Coverage[i-1] = count + block.Count
```

I'm still trying to figure out why gover would produce an invalid coverage (perhaps you have an idea?), but please consider adding some `-debug` info to make this easier to figure out.

See https://github.com/kubernetes/ingress/issues/338 for the background.

**Comment by porridge** [comment]
FTR, https://github.com/modocache/gover/issues/4 explains the cause, and suggests this might be a common issue for others as well.

**Comment by mattn** [comment]
I don't get repro. Below is gover.coverprofile generated.

https://gist.github.com/06be6d12d9c69309dec5b33f62206909

**Comment by gdm85** [comment]
As far as I know this happens when the source files on disk have less line numbers than the generated coverage profile; a sanity check would be in order.

**Comment by gdm85** [comment]
@porridge I am going to close this issue, assuming it is fixed in current `master` now that #180 is merged; if it is not fixed and can still be reproduced (either in this form or in a new one) please comment back.
